app/code/Magento/SalesRule/Model/ResourceModel/Rule.php

```diff
@@ -53,39 +54,33 @@ class Rule extends AbstractResource
      */
     protected $entityManager;
 
-    /**
-     * @var MetadataPool
-     */
-    private $metadataPool;
-
     /**
      * @var string
      */
     private $linkedField;
 
     /**
-     * @param \Magento\Framework\Model\ResourceModel\Db\Context $context
-     * @param \Magento\Framework\Stdlib\StringUtils $string
-     * @param \Magento\SalesRule\Model\ResourceModel\Coupon $resourceCoupon
+     * @param DbContext $context
+     * @param StringUtils $string Magento string lib
+     * @param ResourceCoupon $resourceCoupon
      * @param string $connectionName
-     * @param \Magento\Framework\DataObject|null $associatedEntityMapInstance
+     * @param DataObject|null $associatedEntityMapInstance
      * @param Json $serializer Optional parameter for backward compatibility
      * @param MetadataPool $metadataPool Optional parameter for backward compatibility
      */
     public function __construct(
-        \Magento\Framework\Model\ResourceModel\Db\Context $context,
-        \Magento\Framework\Stdlib\StringUtils $string,
-        \Magento\SalesRule\Model\ResourceModel\Coupon $resourceCoupon,
+        DbContext $context,
+        protected readonly StringUtils $string,
+        ResourceCoupon $resourceCoupon,
```

Comment:
Why not promote this property?
```suggestion
        protected readonly ResourceCoupon $_resourceCoupon,
```